joomla/joomla-cms
# [RFC] uncouple guest_usergroup from com_users

### Summary of Changes

This PR aims to uncomple guest_usergroup param from com_users and add it to global config instead.

### Testing Instructions

- Code review
- Neeeds to be tested on install AND on update
- For install (test several sample data - since they have different guest user groups)
 - Apply patch, delete configuration.php, refresh page, install Joomla
 - Go to global config and make sure the guest usergroup selected is the "Guest" one
 - Try to change the guest user group and confirm its working fine
- For update
 - [TO DO] If positive comments

Note: Use the most recent version of the patchtester https://github.com/joomla-extensions/patchtester/releases/download/3.0.0-alpha2/com_patchtester.zip if you have memory issues installing the patch

### Documentation Changes Required

Any docs that shows you to set the default guest usergroup needs to be updated.

### Notes

- It should be B/C. But please check.

- Besides from a architucture point it seems better this way, this should speed all guest non com_users pages load by 2 to 4 ms and less ~0.2 MB memory because doesn't need to get the param from com_users component. But that can be added again later if other componet uses JComponetHelper (ex: com_languages)
![image](https://cloud.githubusercontent.com/assets/9630530/18609599/b860d28e-7cfd-11e6-974b-d0485f2d99e5.png)

- I didn't test the update process yet, only the new installation process, if comments are favorable to this change will test it and, if needed, make necessary changes.
